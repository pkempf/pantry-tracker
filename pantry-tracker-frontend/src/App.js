import React, { useState } from "react";
import Container from "react-bootstrap/Container";
import UserContext from "./UserContext";
import AlertContext from "./AlertContext";
import NavBar from "./NavBar";
import PantryAlert from "./PantryAlert";
import Routes from "./Routes";
import getHelperFunctions, { useLocalStorage } from "./appHelpers";
import TokenSetter from "./TokenSetter";

function App() {
  const [user, setUser] = useLocalStorage("pantry-user", {});
  const [token, setToken] = useLocalStorage("pantry-token", "");

  const [message, setMessage] = useState({
    text: "",
    variant: "",
  });

  const [logIn, logOut, signUp, editUser] = getHelperFunctions(
    user,
    setUser,
    setMessage,
    setToken
  );

  return (
    <UserContext.Provider value={user}>
      <NavBar />
      <AlertContext.Provider value={{ message, setMessage }}>
        <Container className="mt-3" fluid="lg">
          <PantryAlert />
          <TokenSetter token={token} />
          <Routes
            logIn={logIn}
            logOut={logOut}
            signUp={signUp}
            editUser={editUser}
          />
        </Container>
      </AlertContext.Provider>
    </UserContext.Provider>
  );
}

export default App;
