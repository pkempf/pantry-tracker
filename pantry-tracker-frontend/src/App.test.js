import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import App from "./App";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
});

it("matches the snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
