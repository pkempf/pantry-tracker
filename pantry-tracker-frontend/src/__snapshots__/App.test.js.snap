// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-md navbar-dark bg-primary"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Pantry Tracker
    </a>
    <button
      aria-controls="pantry-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="justify-content-end navbar-collapse collapse"
      id="pantry-navbar-nav"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-link"
          href="/login"
        >
          Login
        </a>
        <a
          class="nav-link"
          href="/signup"
        >
          Sign Up
        </a>
      </div>
    </div>
  </nav>
  <div
    class="mt-3 container-lg"
  >
    <div
      class="mt-3 text-center jumbotron"
    >
      <h1
        style="font-size: 72px;"
      >
        <strong>
          Pantry Tracker
        </strong>
      </h1>
      <h4
        class="my-4"
      >
        Let's get cooking!
      </h4>
      <span>
        <a
          class="btn btn-primary mr-1"
          href="/login"
        >
          Log in
        </a>
        <a
          class="btn btn-primary ml-1"
          href="/signup"
        >
          Sign up
        </a>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
